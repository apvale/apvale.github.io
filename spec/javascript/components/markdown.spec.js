import { mount, shallowMount } from '@vue/test-utils'
import Markdown from '@/components/Markdown'

describe('Markdown', () => {
  let slots, wrapper

  beforeAll(() => {
    wrapper = mount(Markdown, slots)
  })

  describe('with text', () => {
    it('is a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

  describe('with children', () => {
    beforeAll(() => {
      slots = {
        slots: {
          default: '<p>Some text</p>'
        }
      }

      wrapper = mount(Markdown, slots)
    })

    it('is a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })
})
