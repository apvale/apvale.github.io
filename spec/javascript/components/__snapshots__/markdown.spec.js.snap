// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown with children renders correctly 1`] = `
<div>
  <p>Some text</p>
</div>
`;

exports[`Markdown with text renders correctly 1`] = `<div></div>`;
